promql/functions.go

```diff
@@ -1111,11 +1111,17 @@ func funcHistogramStdDev(vals []parser.Value, args parser.Expressions, enh *Eval
 		it := sample.H.AllBucketIterator()
 		for it.Next() {
 			bucket := it.At()
+			if bucket.Count == 0 {
+				continue
+			}
 			var val float64
 			if bucket.Lower <= 0 && 0 <= bucket.Upper {
 				val = 0
 			} else {
 				val = math.Sqrt(bucket.Upper * bucket.Lower)
+				if bucket.Upper < 0 && bucket.Lower < 0 {
```

Comment:
Let's just test for the Upper limit. We have tested the "one is positive, one is negative" a few lines up. And also the math.Sqrt would have resulted in `NaN` otherwise anyway. And finally, if the Upper bucket is negative, than the Lower must be, too. So in many aspects, the following appears more logical and easier to read to me:

```suggestion
				if bucket.Upper < 0 {
```